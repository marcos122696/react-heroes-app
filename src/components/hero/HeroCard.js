import React from 'react'
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { heroImages } from '../../helpers/heroImages';


export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters
}) => {

    // const imgPath = `/assets/${id}.jpg`; public/assets

    return (
        <div className="col">
            <Link to={`/Hero/${id}`} style={{ textDecoration: 'none' }}>
                <div className="card bg-secondary text-white h-100">
                    <div className="row no-gutters">
                        <div className="container-img" >
                            <img src={ heroImages(`./${id}.jpg`)} className="h-100 w-100 animate__animated animate__zoomIn" alt={superhero} />
                        </div>

                        <div className="col-8 w-100">
                            <div className="card-body" >

                                <h5 className="card-title">{superhero}</h5>
                                <p className="card-text">{alter_ego}</p>

                                <p className="card-text">
                                    <small className="text-white-50">{first_appearance}</small>
                                </p>

                            </div>
                        </div>
                    </div>
                </div>
            </Link>
        </div>
    )
};

HeroCard.propTypes = {
    id: PropTypes.string.isRequired,
    superhero: PropTypes.string.isRequired,
    alter_ego: PropTypes.string.isRequired,
    first_appearance: PropTypes.string.isRequired,
};