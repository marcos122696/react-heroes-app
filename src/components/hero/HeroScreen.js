import React, { useMemo } from 'react';
import { useParams, Navigate, useNavigate } from 'react-router-dom';
import { heroImages } from '../../helpers/heroImages';
import { getHeroById } from '../../selectors/getHeroById';



export const HeroScreen = () => {

    const { heroeId } = useParams();

    const hero = useMemo( () => getHeroById(heroeId), [heroeId] );

    const { id, superhero, publisher, alter_ego, first_appearance, characters } = hero;
    
    const navigate = useNavigate();

    const  handleReturn  = () => {
        return navigate(-1);
    };

    if (!hero) {
        return <Navigate to='/' />;
    };

    // const imgPath = `/assets/${ id }.jpg`; public/assets

  return (
    <div className="row mt-5">
        <div className="col-4">
            <img 
                src={ heroImages(`./${id}.jpg`) }
                alt={ superhero }
                className="img-thumbnail animate__animated animate__fadeInLeft"
            />
        </div>

        <div className="col-8">
            <h3>{ superhero }</h3>
            <ul className="list-group list-group-flash">
                <li className="list-group-item list-group-item-primary"> <b>Alter ego:</b> { alter_ego }</li>
                <li className="list-group-item list-group-item-primary"> <b>publisher:</b> { publisher }</li>
                <li className="list-group-item list-group-item-primary"> <b>first Appearance:</b> { first_appearance }</li>
            </ul>

            <h5 className="mt-5">Characters</h5>
            <p>{ characters }</p>

            <button
                className="btn btn-outline-info"
                onClick={ handleReturn }
            >
                Regresar
            </button>

        </div>
    </div>
  )
}
