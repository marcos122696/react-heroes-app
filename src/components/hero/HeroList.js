import React, { useMemo } from 'react'
import { getHeroesByPublisher } from '../../selectors/getHeroesByPublisher'
import { HeroCard } from './HeroCard';
import PropTypes from 'prop-types';

export const HeroList = ({ publisher }) => {
    
    const heroes = useMemo( () => getHeroesByPublisher(publisher), [publisher] );

  return (
    <div className="row rows-cols-1 row-cols-md-3 g-3 text-dark">
        {
            heroes.map( (hero) => (
                <HeroCard 
                    key={ hero.id } 
                    { ...hero}
                />
            ))
        }
    </div>
  )
};

HeroList.propTypes = {
    publisher: PropTypes.string.isRequired,
};
