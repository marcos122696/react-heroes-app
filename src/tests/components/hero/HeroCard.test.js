import { shallow } from "enzyme";
import { HeroCard } from "../../../components/hero/HeroCard";
import { heroes } from "../../../data/heroes";


describe('Pruebas en el <HeroCard />', () => {

    const wrapper = shallow(
        <HeroCard
            id={ heroes[0].id }
            superhero={ heroes[0].superhero }
            alter_ego={ heroes[0].alter_ego }
            first_appearance={ heroes[0].first_appearance }
        />
    );


    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('Debe de haber un h5 con el nombre de Batman', () => {

        expect( wrapper.find('h5').text().trim() ).toBe( 'Batman' );
    });


    test('Debe de aparecer la primera etiqueta "p" con el alter_ego Bruce Wayne', () => {

        expect( wrapper.find('p').at(0).text().trim() ).toBe( 'Bruce Wayne' );
    });

    
    test('Debe de aparecer la primera segunda "p" con el first_appearance Detective Comics #27', () => {

        expect( wrapper.find('p').at(1).text().trim() ).toBe( 'Detective Comics #27' );
    });
});