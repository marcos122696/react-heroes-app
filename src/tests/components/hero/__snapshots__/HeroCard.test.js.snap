// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <HeroCard /> Debe mostrarse correctamente 1`] = `
<div
  className="col"
>
  <Link
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/Hero/dc-batman"
  >
    <div
      className="card bg-secondary text-white h-100"
    >
      <div
        className="row no-gutters"
      >
        <div
          className="container-img"
        >
          <img
            alt="Batman"
            className="h-100 w-100 animate__animated animate__zoomIn"
            src="/assets/dc-batman.jpg"
          />
        </div>
        <div
          className="col-8 w-100"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Batman
            </h5>
            <p
              className="card-text"
            >
              Bruce Wayne
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-white-50"
              >
                Detective Comics #27
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  </Link>
</div>
`;
